sustainability/models/carbon_factor.py

```diff
@@ -0,0 +1,824 @@
+                        carbon.factor.value.id: {
+                            ...
+                        },
+                    },
+                }
+
+        """
+        distribution = self.check_distribution(distribution)
+        total_value = 0.0
+        total_uncertainty_value = 0.0
+        factor_to_details = {}
+        for factor in self:
+            value, uncertainty_value, details = factor._get_carbon_value(
+                distribution.get(factor), **kwargs
+            )
+            total_value += value
+            total_uncertainty_value += uncertainty_value
+            factor_to_details[factor.id] = details
+
+        return total_value, total_uncertainty_value, factor_to_details
+
+    @classmethod
+    def _get_uncertainty_value(
+        cls, uncertainty_percentage: float, data_uncertainty_percentage: float
+    ) -> float:
+        """
+        Return the uncertainty value of a given value depending on the uncertainty percentage. The float is a percentage, 1 = 100%, 0.5 = 50%, etc.
+        """
+        return (uncertainty_percentage**2 + data_uncertainty_percentage**2) ** 0.5
+
+    def _get_physical_effective_quantity(
+        self,
+        factor_value,
+        quantity,
+        from_uom_id,
+        product_id,
+        reference: str | list[str] | None = None,
+    ):
+        """
+        Helper to compute the effective quantity (including weight/UoM conversions) used to multiply by carbon_value for a given factor_value.
+        Returns the effective quantity (float) or raises ValidationError if not computable.
+        Optionally takes a reference (list of strings) to append to error messages for traceability.
+        """
+        if not reference:
+            reference = []
+        if isinstance(reference, str):
+            reference = [reference]
+        if isinstance(reference, list):
+            reference = [str(r) for r in reference if r]
+        ref_str = ""
+        if reference:
+            ref_str = "\nReference: " + "\n- ".join(reference)
+
+        if (
+            factor_value.carbon_compute_method == "physical"
+            and quantity
+            and from_uom_id
+        ):
+            weight_uom_category = self.env.ref("uom.product_uom_categ_kgm")
+            # Case: carbon factor UoM is weight, product UoM is not weight
+            if (
+                self.carbon_uom_id.category_id == weight_uom_category
+                and product_id
+                and product_id.uom_id.category_id != weight_uom_category
+            ):
+                if not product_id.weight or product_id.weight <= 0:
+                    raise ValidationError(
+                        _(
+                            f"The weight may not be defined or is zero for the associated product (%s). Please ensure the weight is properly set to compute the carbon value.{ref_str}",
+                            product_id.display_name,
+                        )
+                    )
```

Comment:
@BonnetAdam 
We have to rework this because uom category does not exist anymore.
Can you check it as you worked on this in the past?
Thanks :)